lobby/src/main/java/org/triplea/lobby/server/db/BannedMacController.java

```diff
@@ -8,26 +8,29 @@
 import java.sql.SQLException;
 import java.sql.Timestamp;
 import java.time.Instant;
+import java.util.function.Supplier;
 
 import javax.annotation.Nullable;
 
 import org.triplea.lobby.server.User;
 import org.triplea.util.Tuple;
 
+import lombok.AllArgsConstructor;
+
 /**
  * Utility class to create/read/delete banned macs (there is no update).
  */
-public class BannedMacController extends TimedController implements BannedMacDao {
-  public BannedMacController(final Database database) {
-    super(database);
-  }
+@AllArgsConstructor
+class BannedMacController implements BannedMacDao {
+
+  private final Supplier<Connection> connection;
 
   @Override
   public void addBannedMac(final User bannedUser, final @Nullable Instant banTill, final User moderator) {
     checkNotNull(bannedUser);
     checkNotNull(moderator);
 
-    if (banTill != null && banTill.isBefore(now())) {
+    if (banTill != null && banTill.isBefore(Instant.now())) {
```

Comment:
Injecting a time provider is not a very good solution, nor is a partial mock. I'm really reluctant to maintain tests that are likely going to be removed soon anyways. There are a number of fixes that we should do, to begin with like using DB `now()` function so we can avoid any clock problems between JVM and the DB engine.

We do have pretty good coverage of these functions already, but I'd agree it's not convincing to drop the tests while waiting for a rewrite. It's an issue though that it's out of scope for this update to merge APIs that would make the testing nicer, yet the testing is duplicated multiple times which multiplies the busy-work of maintaining tests.

Reply:
There's an abnormal number of problems in this code:
(1) we should be using the DB function `now()`, AFAIK that is a best practice to use built-in DB utilities and avoid any clock synchronization issues
(2) the tests are copy/pasts which multiplies the effort
(3) tests are hugely questionable as they depend on previous state. There is repeated entries of the same user, this test setup does not seem valid to begin with
(4) injecting a mock override of now is very questionable, injecting the 'now' time as part of API is a better alternative.
(5) the original APIs are due to be merged up and rewritten, so this test code is going to be deleted.

I took a stab at this and spent the last 30 minutes again fighting with these tests. I think it'll take around 3 hours to update it so we can have minimal changes that support this highly questionable test. Considering I could instead spend that time to rewrite the APIs, and spend the effort towards something with long-term value, I really think maintaining this test is a waste of time and hence the decision to reduce the amount of tests that need to be maintained. Since we have other tests already, the coverage is pretty reasonable.

The question then becomes, are we okay moving forward knowing there will be more progress in this area, or should I attach more changes to this PR where the tests are outright re-written (because the API is likely to change, which means we'll have a lot of other changes as the caller code needs to be updated).

Reply:
Sigh. Leaving lopped off test is not convincing, at the risk of really blowing up the review scope I'm going to loop in refactoring the DB APIs and LobbyLoginValidator testing.

Reply:
this update address this: 27dee9a78ef0fd6496f5aef3a00d28dde127358e

I think this is an impossible condition for expiry time to be in the past, so the 'instant.now' check probably does nothing anyways